.mvn/gradle-enterprise-custom-user-data.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Add Maven command line arguments
+def mavenCommand = ''
+
+if (System.env.MAVEN_CMD_LINE_ARGS) {
+    mavenCommand = "mvn ${System.env.MAVEN_CMD_LINE_ARGS}".toString()
+    buildScan.value('Maven command line', mavenCommand)
```

Comment:
I would be very careful capturing the entire Maven command line. You could imagine scenarios where a user passes in a secret as a system property, or possibly even typos that include secrets. Gradle Enterprise doesn't capture certain environment details for this reason.

Perhaps it would make more sense to capture specific details you may be interested in? For example, we already add a custom value for the `skipTests` setting. We also have samples to [capture profiles](https://github.com/gradle/gradle-enterprise-build-config-samples/blob/main/build-data-capturing-maven-samples/capture-profiles/maven-profiles.groovy) as tags. This may actually make it easier to queries that combine the different values, rather than trying to construct queries that are considering the command line as a whole. It may also be more accurate, as capturing the value of configured properties would not _only_ capture values passed in at the command line.

Reply:
I'll leave it as is for now. If you have better config suggestions, feel free to patch the script based on this branch or after this patch gets merged.

Reply:
> OK. I'm trying to redact mvn command.

@tisonkun - should just be a matter of removing this `if (System.env.MAVEN_CMD_LINE_ARGS) { ... }` block